chips/atmega328p-hal/src/adc.rs

```diff
@@ -18,3 +21,25 @@ avr_hal::impl_adc! {
         }
     }
 }
+
+/// Additional channels
+///
+/// This module contains ADC channels, additional to the direct pin channels.
+pub mod channel {
+    use avr_hal::hal::adc::Channel;
+    use crate::atmega328p::adc::admux::MUX_A;
+
+    /// Channel for the _Bandgap Reference Voltage_
+    pub struct Vbg;
+    impl Channel<super::Adc> for Vbg {
+        type ID = MUX_A;
+        fn channel() -> Self::ID { MUX_A::ADC_VBG }
+    }
+
+    /// Channel for _GND_
+    pub struct Gnd;
+    impl Channel<super::Adc> for Gnd {
+        type ID = MUX_A;
+        fn channel() -> Self::ID { MUX_A::ADC_GND }
+    }
+}
```

Comment:
I looks like you forgot to add the Temperature Channel for the atmega328p

Reply:
Ack, I wasn't entirely sure how the temperature sensor works on ATmega328P.  I'll have to take another look at the datasheet.